modules/simple-payments/simple-payments.php

```diff
@@ -27,26 +27,34 @@ static function getInstance() {
 		return self::$instance;
 	}
 
-	private function register_scripts() {
+	private function register_scripts_and_styles() {
 		/**
 		 * Paypal heavily discourages putting that script in your own server:
 		 * @see https://developer.paypal.com/docs/integration/direct/express-checkout/integration-jsv4/add-paypal-button/
 		 */
 		wp_register_script( 'paypal-checkout-js', 'https://www.paypalobjects.com/api/checkout.js', array(), null, true );
 		wp_register_script( 'paypal-express-checkout', plugins_url( '/paypal-express-checkout.js', __FILE__ ),
 			array( 'jquery', 'paypal-checkout-js' ), self::$version );
+		wp_register_style( 'jetpack-simple-payments', plugins_url( '/simple-payments.css', __FILE__ ), array( 'dashicons' ) );
 	}
+
 	private function register_init_hook() {
 		add_action( 'init', array( $this, 'init_hook_action' ) );
 	}
+
 	private function register_shortcode() {
 		add_shortcode( self::$shortcode, array( $this, 'parse_shortcode' ) );
 	}
 
 	public function init_hook_action() {
+		if ( ! $this->is_enabled_jetpack_simple_payments() ) {
+			add_shortcode( self::$shortcode, array( $this, 'ignore_shortcode' ) );
```

Comment:
Can we call `remove_shortcode` instead here? It's always a little strange to have a `noop` method 🙃

Reply:
The problem is that `remove_shortcode` just removes the hooks associated with the shortcode, but it will still show the shortcode as content.

| `remove_shortcode` | `add_shortcode(..., noop)`|
| - | - |
| ![screen shot 2018-05-31 at 00 12 27](https://user-images.githubusercontent.com/233601/40759246-9312edb6-6467-11e8-973f-d542d483dc6e.png) | ![screen shot 2018-05-31 at 00 13 12](https://user-images.githubusercontent.com/233601/40759253-9d134356-6467-11e8-9c0e-add9eb1c779f.png) |

My intention is not to break the site showing a shortcode as text.